src/main/java/org/wildfly/security/authz/ConstantPermissionMapper.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.authz;
+
+import static org.wildfly.security._private.ElytronMessages.log;
+import static org.wildfly.common.Assert.checkNotNullParam;
+
+import org.wildfly.security.permission.PermissionVerifier;
+
+/**
+ * A simple {@link PermissionMapper} implementation that maps all to pre-defined {@link PermissionVerifier} instance.
+ *
+ * This {@code PermissionMapper} is constructed using a {@link Builder}.
+ *
+ * @author <a href="mailto:jkalina@redhat.com">Jan Kalina</a>
+ */
+public class ConstantPermissionMapper implements PermissionMapper {
```

Comment:
This all seems like overkill to me.  Why not just add one method to ``PermissionMapper``:

```
static PermissionMapper createConstant(PermissionVerifier verifier) {
    return (p, r) -> verifier;
}
```